Load app colors from CSS themes

gerhardol opened the pull request:
Fixes #9782

## Proposed changes

Restore the possibility to change application colors from .css files.
While the theme handling is unfortunately disabled after .NET5, the possibility to set colors like branches was available before theming and is expected by users.

@NikolayXHD suggested that ExCSS is removed and replaced with regex parsing, to get rid of a dependency and get parse errors (that must be ignored with ExCSS >= 3).
This is not done due to simplicity, easier to fix this than to reimplement.  (If the implementation is limited to allow only single line comments, one command per line etc, it would be simpler and maybe feasible.)

## Screenshots <!-- Remove this section if PR does not change UI -->

### Before

![image](https://user-images.githubusercontent.com/6248932/147742823-53ca1af2-510e-449b-a82e-803c604bb3f5.png)

### After

Weird mix of switch local-remote colors and colorblind
![image](https://user-images.githubusercontent.com/6248932/147742946-14abf7f1-e256-4184-90ac-be3c9e72b353.png)

## Test methodology <!-- How did you ensure quality? -->

Renabled and adapted tests

## Merge strategy

I agree that the maintainer squash merge this PR (if the commit message is clear).

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

RussKie commented:
👍

[nit] I would skip/ignore the test instead of the pragma.

On Thu, 30 Dec 2021, 9:38 pm NikolayHD, ***@***.***> wrote:

> ***@***.**** approved this pull request.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/gitextensions/gitextensions/pull/9784#pullrequestreview-841829302>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABBTEXRRTTL3MDZ657UQCITUTQZBNANCNFSM5K7O4OJA>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

gerhardol commented:
Updated comment about no errors parsed (and that this could be rewritten), removed the dummy rule and unnecessary handling.
Plan to merge tomorrow.

Issue about the error parsing in ExCSS:
https://github.com/TylerBrinks/ExCSS/issues/132
